storage/client_merge_test.go

```diff
@@ -265,3 +265,74 @@ func TestStoreRangeMergeNonConsecutive(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestStoreRangeMergeWithSpanRangeScan simulate a bug: A spanning-range scan may get
+// dumplicate key-values when two ranges are mergeing.
```

Comment:
What is this test actually testing? It doesn't use a `DistSender` after the merge. If we can't reproduce the duplicate result bug then I don't think it makes sense to add this test as it is.